src/main/java/ch/njol/skript/events/EvtWorldSave.java

```diff
@@ -0,0 +1,66 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import org.bukkit.World;
+import org.bukkit.event.Event;
+import org.bukkit.event.world.WorldSaveEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Arrays;
+
+public class EvtWorldSave extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("World Save", EvtWorldSave.class, WorldSaveEvent.class, "world sav(e|ing) [of %-worlds%]")
+			.description("Called when a world is saved to disk. Usually all worlds are saved simultaneously, but world management plugins could change this.")
+			.examples(
+				"on world save of \"world\":",
+				"\tbroadcast \"The world %event-world% has been saved\"")
+			.since("1.0, INSERT VERSION (defining worlds)");
+	}
+
+	@Nullable
+	private World[] worlds;
+
+	@Override
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
+		if (worlds != null)
+			worlds = ((Literal<World>) args[0]).getArray();
```

Comment:
I stated this for the other event, but worlds is always going to be null so there is no point of the if (worlds != null) statement.

Reply:
Since removing this in events I've received this error from it
``java.lang.NullPointerException: Cannot invoke "ch.njol.skript.lang.Literal.getArray()" because "args[0]" is null``

this occurs when it's `on world save:`